src/pipx/venv.py

```diff
@@ -184,19 +263,130 @@ def get_python_version(self) -> str:
             .strip()
         )
 
+    def pip_search(self, search_term, pip_search_args):
+        cmd = [str(self.python_path), "-m", "pip", "search"]
+        cmd += pip_search_args + [search_term]
+        cmd_run = subprocess.run(cmd, stdout=subprocess.PIPE)
+        return cmd_run.stdout.decode().strip()
+
+    def list_installed_packages(self) -> Set[str]:
+        cmd = [str(self.python_path), "-m", "pip", "list", "--format=json"]
+        cmd_run = subprocess.run(cmd, stdout=subprocess.PIPE)
+        pip_list = json.loads(cmd_run.stdout.decode().strip())
+        return set([x["name"] for x in pip_list])
+
+    def top_of_deptree(self, packages):
```

Comment:
add type annotations

Reply:
Should have a unit test added for this function.

Reply:
Added a unit test in test_run for package name determination.  See if that looks good.

I'm afraid it might fail without the merge of #276